src/core/server/http/http_server.ts

```diff
@@ -148,15 +148,18 @@ export class HttpServer {
         this.log.debug(`registering route handler for [${route.path}]`);
         // Hapi does not allow payload validation to be specified for 'head' or 'get' requests
         const validate = ['head', 'get'].includes(route.method) ? undefined : { payload: true };
-        const { authRequired = true, tags, body = {} } = route.options;
+        const { authRequired = true, tags, body = {}, xsrfRequired = true } = route.options;
         const { accepts: allow, maxBytes, output, parse } = body;
+        const kibanaRouteState: KibanaRouteState = { xsrfRequired };
+
         this.server.route({
           handler: route.handler,
           method: route.method,
           path: route.path,
           options: {
             // Enforcing the comparison with true because plugins could overwrite the auth strategy by doing `options: { authRequired: authStrategy as any }`
             auth: authRequired === true ? undefined : false,
+            app: kibanaRouteState,
```

Comment:
NIT: can't this be inlined to remove the local var?

Reply:
We can, but I want to make sure that `kibanaRouteState` implements KibanaRouteState correctly. 
This declaration doesn't show an error for unknown properties, for example:
```typescript
app: { xsrfRequired } as KibanaRouteState
```